src/tribol/mesh/CouplingScheme.cpp

```diff
@@ -1708,29 +1765,91 @@ void CouplingScheme::updatePairReportingData( const FaceGeomError face_error )
       default: break;
    } // end switch
 }
+
 //------------------------------------------------------------------------------
 void CouplingScheme::printPairReportingData()
 {
-   int numInterfacePairs = this->m_interfacePairs->getNumPairs();
-
-   SLIC_DEBUG(this->m_numActivePairs*100./numInterfacePairs << "% of binned interface " <<
+   SLIC_DEBUG(this->getNumActivePairs()*100./getInterfacePairs().size() << "% of binned interface " <<
               "pairs are active contact candidates.");
 
    SLIC_DEBUG_IF(this->m_pairReportingData.numBadOrientation>0,
                  "Number of bad orientations is " << this->m_pairReportingData.numBadOrientation <<
-                 " equaling " << this->m_pairReportingData.numBadOrientation*100./numInterfacePairs <<
+                 " equaling " << this->m_pairReportingData.numBadOrientation*100./getInterfacePairs().size() <<
                  "% of total number of binned interface pairs.");
 
    SLIC_DEBUG_IF(this->m_pairReportingData.numBadFaceGeometry>0,
                  "Number of bad face geometries is " << this->m_pairReportingData.numBadFaceGeometry <<
-                 " equaling " << this->m_pairReportingData.numBadFaceGeometry*100./numInterfacePairs <<
+                 " equaling " << this->m_pairReportingData.numBadFaceGeometry*100./getInterfacePairs().size() <<
                  "% of total number of binned interface pairs.");
 
    SLIC_DEBUG_IF(this->m_pairReportingData.numBadOverlaps>0,
                  "Number of bad contact overlaps is " << this->m_pairReportingData.numBadOverlaps <<
-                 " equaling " << this->m_pairReportingData.numBadOverlaps*100./numInterfacePairs <<
+                 " equaling " << this->m_pairReportingData.numBadOverlaps*100./getInterfacePairs().size() <<
                  "% of total number of binned interface pairs.");
 }
+
+//------------------------------------------------------------------------------
+template <typename T, typename PARAM, typename MESH, typename CP, typename CP2D, typename CP3D>
+CouplingScheme::ViewerBase<T, PARAM, MESH, CP, CP2D, CP3D>::ViewerBase( T& cs )
+  : m_parameters( cs.m_parameters )
+  , m_contact_case( cs.m_contactCase )
+  , m_enforcement_options( cs.m_enforcementOptions )
+  , m_mesh1( cs.getMesh1() )
```

Comment:
This may change if we move to the coupling scheme holding pointers to mesh data. If we need the view here let's name the routine `getMeshView1()`.